src/strpkg/extract.nim

```diff
@@ -176,6 +178,14 @@ proc adjust_by*(A:var tread, B:tread, opts:Options, B_position:uint32): bool =
         A.position = B_position + B.align_length.uint32
 
     # if we have adjusted A based on B, then any split is not informative.
+    if A.split != Soft.none:
+      # the split affects the align-length. we no longer have the original
+      # length, but we can set it to the max, re-check p_repeat and skip if
+      # it's not high enough.
```

Comment:
or this section

Reply:
we can remove this section after dropping the section in get_repeat that drops soft-clips.